### PR TITLE
Added feedback messages for successful execution of commands

### LINKED ISSUE #744: Add feedback messaging to plugin repo, plugin install
(This is used to request new product features)

**Describe the feature request**
During the TAP beta 2 bug bash we found that folks were sometimes uncertain if commands had executed successfully when there was no feedback. 

Specific examples included the `tanzu plugin repo update -b tanzu-cli-framework core` and `tanzu plugin install --local cli all` commands which do not give feedback when successful. 

# Ideas:
## Plugin repo update
```
$tanzu plugin repo update -b tanzu-cli-framework core
Plugin repo updated
$
```

## Plugin install
```
$tanzu plugin install --local cli all
installing plugin apps v0.5.0 ... OK
installing plugin package v1.4.0  ... OK
$
```
* might be less heavy than the table for single plugin installs
or
```
$tanzu plugin install --local cli all
Installed plugins
NAME      DESCRIPTION                  VERSION
apps      Applications on Kubernetes   v0.5.0
package   Tanzu Package Management     v0.5.0
$
```
* might be a bit much when installing a single plugin, but nice for bulk installs (not sure how often that happens after day 1)

**Describe alternatives you've considered**
```
$tanzu plugin install --local cli all
OK
$
```

```
$tanzu plugin install --local cli all
Success
$
```
```
$tanzu plugin install --local cli all
Complete
$
```

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [ ] Addons
- [x] CLI
- [ ] Docs
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [x] User Experience

**Additional context**

### PR DESCRIPTION
**Pull request by saji-pivotal**
Added feedback messages for successful execution of plugin clean and plugin repo commands

### What this PR does / why we need it
This PR enhances the UX of the Tanzu CLI

### Which issue(s) this PR fixes
Fixes #744

### Describe testing done for PR
1. Built the code with the changes
2. Ran the following commands
```
tanzu plugin repo add -n core -b tanzu-cli-framework -p gcr.io
✔  successfully added repository core

tanzu plugin repo update -b tanzu-cli-framework core
✔  Successfully updated repository configuration for core

tanzu plugin repo delete core
✔  Successfully deleted repository core

tanzu plugin clean
✔  Successfully cleaned up all plugins
```

### Release note

```release-note
Improved UX of the Tanzu CLI by adding success manages if commands complete successfully
```

### CODE REVIEW

**Review by marckhouzam — COMMENTED** [review]
This looks good.

While at it, do you want to do the `plugin source` commands as well?  

Also, do we still actively support the legacy plugin discovery?  I'm asking because looking at the code I believe there is no success message for `plugin upgrade` or `plugin delete` for the legacy discovery case.

**Review by mpanchajanya — APPROVED** [review]

File: pkg/v1/cli/command/core/discovery_source.go [hunk 1]
```diff
@@ -129,6 +131,7 @@ var addDiscoverySourceCmd = &cobra.Command{
 		if err != nil {
 			return err
 		}
+		log.Successf("Successfully added discovery sources")
```
Comment (marckhouzam) [inline pkg/v1/cli/command/core/discovery_source.go:134]: Could we be adding more than one source?  If not maybe "sources" -> "source"

File: pkg/v1/cli/command/core/discovery_source.go [hunk 2]
```diff
@@ -175,6 +178,7 @@ var updateDiscoverySourceCmd = &cobra.Command{
 		if err != nil {
 			return err
 		}
+		log.Successf("Updated discover source %s", discoveryName)
```
Comment (marckhouzam) [inline pkg/v1/cli/command/core/discovery_source.go:181]: "discover" -> "discovery"

File: pkg/v1/cli/command/core/discovery_source.go [hunk 3]
```diff
@@ -211,6 +215,7 @@ var deleteDiscoverySourceCmd = &cobra.Command{
 		if err != nil {
 			return err
 		}
+		log.Successf("Deleted discover source %s", discoveryName)
```
Comment (marckhouzam) [inline pkg/v1/cli/command/core/discovery_source.go:218]: "discover" -> "discovery"

File: pkg/v1/cli/command/core/plugin_manager.go [hunk 4]
```diff
@@ -366,7 +366,11 @@ var upgradePluginCmd = &cobra.Command{
 
 		versionSelector := repo.VersionSelector()
 		err = cli.UpgradePlugin(pluginName, plugin.FindVersion(versionSelector), repo)
-		return
+		if err != nil {
+			return err
+		}
+		log.Successf("successfully upgraded plugin %s", pluginName)
```
Comment (marckhouzam) [inline pkg/v1/cli/command/core/plugin_manager.go:372]: Nit: to be consistent with the other messages, "successfully" -> "Successfully" (with a capital letter)

Oh but now I see that the existing message in this function does not do that...
@chandrareddyp opened #1945 about such inconsistencies .

File: pkg/v1/cli/command/core/plugin_manager.go [hunk 5]
```diff
@@ -402,8 +406,11 @@ var deletePluginCmd = &cobra.Command{
 		}
 
 		err = cli.DeletePlugin(pluginName)
-
-		return
+		if err != nil {
+			return err
+		}
+		log.Successf("successfully deleted plugin %s", pluginName)
```
Comment (marckhouzam) [inline pkg/v1/cli/command/core/plugin_manager.go:412]: Here too, the casing is inconsistent (copied from the message higher up).

**Review by marckhouzam — APPROVED** [review]
Thanks for addressing previous comments.

This looks fine to me with some nits about consistency with upper/lower case, which was broken before your PR...  Up to you to decide what you prefer to do about it.

File: pkg/v1/cli/command/core/plugin_manager.go (hunk 4 above)
Comment (chandrareddyp) [reply pkg/v1/cli/command/core/plugin_manager.go:372]: I think the guideline is - "every line should start with a small letter only", so we are good here.

File: pkg/v1/cli/command/core/repo.go [hunk 6]
```diff
@@ -104,6 +106,7 @@ var addRepoCmd = &cobra.Command{
 		if err != nil {
 			return err
 		}
+		log.Successf("Successfully added repository %s", pluginRepo)
```
Comment (chandrareddyp) [inline pkg/v1/cli/command/core/repo.go:109]: start the message with a small letter "successfully" to be consistent with other messages.

File: pkg/v1/cli/command/core/repo.go [hunk 7]
```diff
@@ -152,6 +155,7 @@ var updateRepoCmd = &cobra.Command{
 		if err != nil {
 			return err
 		}
+		log.Successf("Successfully updated repository configuration for %s", repoName)
```
Comment (chandrareddyp) [inline pkg/v1/cli/command/core/repo.go:158]: same here...start the message with a small letter "successfully" to be consistent with other messages.

File: pkg/v1/cli/command/core/repo.go [hunk 8]
```diff
@@ -191,6 +195,7 @@ var deleteRepoCmd = &cobra.Command{
 		if err != nil {
 			return err
 		}
+		log.Successf("Successfully deleted repository %s", repoName)
```
Comment (chandrareddyp) [inline pkg/v1/cli/command/core/repo.go:198]: same here...start the message with a small letter "successfully" to be consistent with other messages.

File: pkg/v1/cli/command/core/plugin_manager.go (hunk 4 above)
Comment (saji-pivotal) [reply pkg/v1/cli/command/core/plugin_manager.go:372]: @marckhouzam I remember that while logging errors, we want to start the sentence with a small letter. I'll edit all success messages in this PR to start with a small letter

**Review by anujc25 — APPROVED** [review]

**Pull request merged**